src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -7172,34 +7173,40 @@ private bool IsEndOfFixedStatement()
                 || this.CurrentToken.Kind == SyntaxKind.SemicolonToken;
         }
 
+        // ParseEmbeddedStatement is called through many recursive statement parsing cases. We want
+        // to ensure it is inlined into the callers as the overhead of this single method can have a
+        // deep impact on the number of recursive calls we can make (more than a hundred during
+        // empirical testing).
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
Is the inlining, or lack of inlining, of the other methods critical to this fix? If so we should probably label them all as inline / don't inline to guard against subtle changes in runtime inlining decisions.

Reply:
Done!